server/src/test/java/org/opensearch/gateway/remote/RemoteClusterStateServiceTests.java

```diff
@@ -658,6 +737,991 @@ public void testWriteIncrementalMetadataSuccessWhenPublicationEnabled() throws I
+            .put("cluster.remote_store.state.global_metadata.upload_timeout", writeTimeout + "s")
+            .build();
+        clusterSettings.applySettings(newSettings);
+        remoteClusterStateService.start();
+        RemoteStateTransferException exception = assertThrows(
+            RemoteStateTransferException.class,
+            () -> remoteClusterStateService.writeMetadataInParallel(
+                ClusterState.EMPTY_STATE,
+                emptyList(),
+                emptyMap(),
+                emptyMap(),
+                true,
+                true,
+                true,
+                true,
+                true,
+                true,
+                emptyMap(),
+                true,
+                emptyList()
+            )
+        );
+        assertTrue(exception.getMessage().startsWith("Timed out waiting for transfer of following metadata to complete"));
+    }
+
+    public void testGetClusterStateForManifest_IncludeEphemeral() throws IOException {
+        ClusterMetadataManifest manifest = generateClusterMetadataManifestWithAllAttributes().build();
+        mockBlobStoreObjects();
+        remoteClusterStateService.start();
+        RemoteReadResult mockedResult = mock(RemoteReadResult.class);
+        RemoteIndexMetadataManager mockedIndexManager = mock(RemoteIndexMetadataManager.class);
+        RemoteGlobalMetadataManager mockedGlobalMetadataManager = mock(RemoteGlobalMetadataManager.class);
+        RemoteClusterStateAttributesManager mockedClusterStateAttributeManager = mock(RemoteClusterStateAttributesManager.class);
+        remoteClusterStateService.setRemoteIndexMetadataManager(mockedIndexManager);
+        remoteClusterStateService.setRemoteGlobalMetadataManager(mockedGlobalMetadataManager);
+        remoteClusterStateService.setRemoteClusterStateAttributesManager(mockedClusterStateAttributeManager);
+        ArgumentCaptor<LatchedActionListener<RemoteReadResult>> listenerArgumentCaptor = ArgumentCaptor.forClass(
+            LatchedActionListener.class
+        );
+        when(mockedIndexManager.getAsyncIndexMetadataReadAction(any(), anyString(), listenerArgumentCaptor.capture())).thenReturn(
+            () -> listenerArgumentCaptor.getValue().onResponse(mockedResult)
+        );
+        when(mockedGlobalMetadataManager.getAsyncMetadataReadAction(any(), anyString(), listenerArgumentCaptor.capture())).thenReturn(
+            () -> listenerArgumentCaptor.getValue().onResponse(mockedResult)
+        );
+        when(mockedClusterStateAttributeManager.getAsyncMetadataReadAction(anyString(), any(), listenerArgumentCaptor.capture()))
+            .thenReturn(() -> listenerArgumentCaptor.getValue().onResponse(mockedResult));
+        when(mockedResult.getComponent()).thenReturn(COORDINATION_METADATA);
+        RemoteClusterStateService mockService = spy(remoteClusterStateService);
+        mockService.getClusterStateForManifest(ClusterName.DEFAULT.value(), manifest, NODE_ID, true);
```

Comment:
Shouldnt you verify the ClusterState returned from this method ? Or is this being covered in other tests

Reply:
Yes, just asserting on the readClusterStateInParallel method call here. We have already tested the readClusterStateInParallel method response in another test.